pkg/common/consumer/consumer_handler.go

```diff
@@ -52,13 +54,24 @@ func WithSaramaConsumerLifecycleListener(listener SaramaConsumerLifecycleListene
 	}
 }
 
+func noopOffsetInitializer(sarama.ConsumerGroupSession) error {
+	return nil
+}
+
+func WithOffsetInitializer(initializer func(sarama.ConsumerGroupSession) error) SaramaConsumerHandlerOption {
```

Comment:
Can you add a comment explaining where this is invoked? And why can't you just use the lifecycle listener instead?

Reply:
I can't recall exactly why. Maybe because the lifecyclelistener does return an error. Anyway I changed it as suggested.

Reply:
if you need to, change the signature to return an error